File: src/tripweaver/domain/schemas/share.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class CreateShareRequest(BaseModel):
    """创建分享链接请求"""
    itinerary_id: int
    expire_days: int | None = 7  # 默认7天有效期，0表示永久


class CreateShareResponse(BaseModel):
    """创建分享链接响应"""
    share_id: str
    share_url: str
    expire_at: datetime | None
    created_at: datetime


class ItineraryDetailResponse(BaseModel):
    """行程详情响应"""
    itinerary_id: int
    destination: str
    days: int
    interests: list[str]
    plan_results: Any
    is_locked: bool
    locked_at: datetime | None
    final_plan_index: int | None
    creator_id: int
    creator_nickname: str
    created_at: datetime
    updated_at: datetime
    members: list[dict] = []
    vote_stats: list[dict] = []
    current_user_vote: int | None = None
    is_creator: bool = False

File: frontend/src/types/share.ts
export interface CreateShareRequest {
  itinerary_id: number
  expire_days?: number
}

export interface CreateShareResponse {
  share_id: string
  share_url: string
  expire_at: string | null
  created_at: string
}

export interface Member {
  user_id: number
  nickname: string
  avatar: string | null
  joined_at: string
}

export interface VoteStat {
  plan_index: number
  count: number
}

export interface ItineraryDetailResponse {
  itinerary_id: number
  destination: string
  days: number
  interests: string[]
  plan_results: any
  is_locked: boolean
  locked_at: string | null
  final_plan_index: number | null
  creator_id: number
  creator_nickname: string
  created_at: string
  updated_at: string
  members: Member[]
  vote_stats: VoteStat[]
  current_user_vote: number | null
  is_creator: boolean
}
